unity-renderer/Assets/UIComponents/Scripts/Components/BaseComponentView.cs

```diff
@@ -50,12 +50,19 @@ public abstract class BaseComponentView : MonoBehaviour, IBaseComponentView
 
     internal ShowHideAnimator showHideAnimator;
 
-    public virtual void Initialize()
+    internal void Initialize()
     {
         isInitialized = false;
         showHideAnimator = GetComponent<ShowHideAnimator>();
+
+        PostInitialization();
+
+        isInitialized = true;
```

Comment:
Love the refactor!

Just double checking here. Is it intended that we call `PostInitialization` before setting the flag as true?

Reply:
Yes, it is intended. The initialized flag/event is maked to be used, for example, by UI components which, in turn, use other UI components as children and maybe it need to know when they have been fully initialized.

Anyway, to improve the code understanding, I have changed the names for this flag/event: now they ara called as `isFullyInitialized` and `OnFullyInitialized`. Done in: https://github.com/decentraland/unity-renderer/pull/1175/commits/55dfd55a8484c8dec18beae6d435a77ae17594fb